.gitlab-ci.yml

```diff
@@ -0,0 +1,62 @@
+stages:
+  - test
+  - deploy
+
+test:
+  stage: test
+  image: python:3.8.7-slim-buster
+  before_script:
+    - python -m pip install --upgrade pip
+    - pip install pytest certifi
+    - if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - python setup.py install
+    # reduce the default SSL security level to get around a misconfiguration in the Ensembl server
+    # https://github.com/Ensembl/ensembl-rest/issues/427
+    # See https://askubuntu.com/a/1233456
+    - sed -i 's/DEFAULT@SECLEVEL=2/DEFAULT@SECLEVEL=1/' /usr/lib/ssl/openssl.cnf
+  script:
+    - PYTHONPATH=. pytest tests
+  environment:
+    name: test-env
+  only:
+    - external_pull_requests
+    - tags
+    - master
+
+
+# Not executed, parent job definition
+.deploy-python-tool:
+  stage: deploy
+  image: python:3.8.7-slim-buster
+  environment:
+    name: $ENVIRONMENT_NAME
+  script:
+    # Start ssh-agent and set identity then copy the Jar file and create a symlink
+    # See https://docs.gitlab.com/ee/ci/ssh_keys/
+    - apt-get update
+    - apt-get -y install openssh-client
+    - eval "$(ssh-agent)"
+    - echo "$EVA_ETL_PRIVATE_KEY" | tr -d '\r' | ssh-add -
+    - ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ${CLUSTER_USER}@${CLUSTER_LOGIN_NODE} "${EVA_SUBMISSION_PATH}/deploy.sh ${DEPLOY_POINT}"
+  dependencies:
+    - test
+
+deploy-python-tool-development:
+  extends: .deploy-python-tool
+  variables:
+    ENVIRONMENT_NAME: development
+    EVA_SUBMISSION_PATH: $EVA_SUBMISSION_DEVELOPMENT_PATH
+    DEPLOY_POINT: $CI_COMMIT_BRANCH
+  only:
+    - external_pull_requests
+    - master
```

Comment:
Does this mean the dev deployment job is run both when a pull request is made and when it's merged to master?  I'm not sure I understand the reason for this.

Reply:
Yes that's correct.
The idea is that the `external_pull_requests` deploy the development branches and the `master` deploys the master branch every time it is merged. 
Because the development deployment is done under the branch name it means we will always have the latest master deployed.
Note that `external_pull_requests` only works on PRopen from `EBIvariation` not from forks.

Reply:
OK makes sense, I think I'm still just used to development without using forks, in which case deploying on PRs and merges might be a lot... thanks for clarifying!